### ISSUE TITLE
[forestdb] All items not replicating between device and sync_gateway

### ISSUE THREAD

**Issue by sethrosetter** [couchbase/couchbase-lite-java-core#965]
Running 1.2.0-sqlcipher (d3b2164c53fb304802f80ec661a8ded4baa54efd)
![device-2016-01-28-234457](https://cloud.githubusercontent.com/assets/1365736/12669934/302a7b0e-c619-11e5-8898-814e31bd03ad.png)

Running 1.2.0-forestdb (d3b2164c53fb304802f80ec661a8ded4baa54efd)
![screenshot 2016-01-28 23 45 59](https://cloud.githubusercontent.com/assets/1365736/12669955/50151ef6-c619-11e5-823f-6a39541fcc02.png)

sync_gateway has all documents
```
curl -X GET http://localhost:4985/todos/_all_docs\?include_docs\=true
```
```
{"rows":[
{"key":"3fb87cd2-cf2b-4fb7-92bb-3cf065197f1a","id":"3fb87cd2-cf2b-4fb7-92bb-3cf065197f1a","value":{"rev":"4-346675cfe27f7c46f30db778f8e43d47"},"doc":{"_attachments":{"image":{"content_type":"image/jpg","digest":"sha1-fft/uCxHSnW9x6uzR/l+bT6R6kQ=","length":17433,"revpos":4,"stub":true}},"_id":"3fb87cd2-cf2b-4fb7-92bb-3cf065197f1a","_rev":"4-346675cfe27f7c46f30db778f8e43d47","checked":false,"created_at":"2016-01-28T23:14:20.129Z","list_id":"c2ce68da-7581-411c-8c07-04cc9f3d2169","title":"sj1","type":"task"}}
,{"key":"6793bf76-9e55-4970-8d3a-9419d65d33fd","id":"6793bf76-9e55-4970-8d3a-9419d65d33fd","value":{"rev":"2-4d8b5b44399548f1683157c62c58c10f"},"doc":{"_id":"6793bf76-9e55-4970-8d3a-9419d65d33fd","_rev":"2-4d8b5b44399548f1683157c62c58c10f","checked":true,"created_at":"2016-01-28T23:19:27.736Z","list_id":"6ec9a50e-dcea-40a3-add2-8b7636372dad","title":"t1","type":"task"}}
,{"key":"6ec9a50e-dcea-40a3-add2-8b7636372dad","id":"6ec9a50e-dcea-40a3-add2-8b7636372dad","value":{"rev":"2-0cf14a99609e602cb64869200cfe6588"},"doc":{"_id":"6ec9a50e-dcea-40a3-add2-8b7636372dad","_rev":"2-0cf14a99609e602cb64869200cfe6588","created_at":"2016-01-28T23:13:44.878Z","members":["p:102664883425143"],"owner":"profile:106418473048384","title":"devicelist","type":"list"}}
,{"key":"7b405bd1-1c5b-4738-9e43-cc80cf64f962","id":"7b405bd1-1c5b-4738-9e43-cc80cf64f962","value":{"rev":"2-c17559eaf4f41c305155b0ed576fada9"},"doc":{"_id":"7b405bd1-1c5b-4738-9e43-cc80cf64f962","_rev":"2-c17559eaf4f41c305155b0ed576fada9","checked":true,"created_at":"2016-01-28T23:14:14.011Z","list_id":"6ec9a50e-dcea-40a3-add2-8b7636372dad","title":"d2","type":"task"}}
,{"key":"93477fad-cdea-4fae-af94-86e9e3630f49","id":"93477fad-cdea-4fae-af94-86e9e3630f49","value":{"rev":"5-c68e078bf3f1815353f0eb5fa3f71b10"},"doc":{"_id":"93477fad-cdea-4fae-af94-86e9e3630f49","_rev":"5-c68e078bf3f1815353f0eb5fa3f71b10","checked":false,"created_at":"2016-01-28T23:13:51.450Z","list_id":"6ec9a50e-dcea-40a3-add2-8b7636372dad","title":"d1","type":"task"}}
,{"key":"b4a15ccb-0642-4778-a921-edbc79912d2d","id":"b4a15ccb-0642-4778-a921-edbc79912d2d","value":{"rev":"1-73018ffb9686f4fc169e07b88a3ccb67"},"doc":{"_id":"b4a15ccb-0642-4778-a921-edbc79912d2d","_rev":"1-73018ffb9686f4fc169e07b88a3ccb67","checked":false,"created_at":"2016-01-28T23:19:29.311Z","list_id":"6ec9a50e-dcea-40a3-add2-8b7636372dad","title":"t3","type":"task"}}
,{"key":"c2ce68da-7581-411c-8c07-04cc9f3d2169","id":"c2ce68da-7581-411c-8c07-04cc9f3d2169","value":{"rev":"2-80eac949599be8343cf7af32925ef68b"},"doc":{"_id":"c2ce68da-7581-411c-8c07-04cc9f3d2169","_rev":"2-80eac949599be8343cf7af32925ef68b","created_at":"2016-01-28T23:13:26.103Z","members":["p:106418473048384"],"owner":"profile:102664883425143","title":"Emulist","type":"list"}}
,{"key":"cefc7081-cdfa-4008-875e-8fa368997c5c","id":"cefc7081-cdfa-4008-875e-8fa368997c5c","value":{"rev":"1-77a62f6a1afabc8c48a1b86e89814fbc"},"doc":{"_id":"cefc7081-cdfa-4008-875e-8fa368997c5c","_rev":"1-77a62f6a1afabc8c48a1b86e89814fbc","checked":false,"created_at":"2016-01-28T23:27:34.501Z","list_id":"6ec9a50e-dcea-40a3-add2-8b7636372dad","title":"d6","type":"task"}}
,{"key":"d4013f29-80a8-41f7-85c5-7f6d6f3a18fe","id":"d4013f29-80a8-41f7-85c5-7f6d6f3a18fe","value":{"rev":"2-e9c46f07ace2253ed44963982db6c68f"},"doc":{"_id":"d4013f29-80a8-41f7-85c5-7f6d6f3a18fe","_rev":"2-e9c46f07ace2253ed44963982db6c68f","checked":true,"created_at":"2016-01-28T23:14:22.430Z","list_id":"c2ce68da-7581-411c-8c07-04cc9f3d2169","title":"jksl4","type":"task"}}
,{"key":"p:102664883425143","id":"p:102664883425143","value":{"rev":"1-9d0f8c07b9fc42a7cf9c00ca9eda7020"},"doc":{"_id":"p:102664883425143","_rev":"1-9d0f8c07b9fc42a7cf9c00ca9eda7020","name":"Seth Two","type":"profile","user_id":"102664883425143"}}
,{"key":"p:106418473048384","id":"p:106418473048384","value":{"rev":"1-81f88fc9259f481c97f60e8c3a8ab4a3"},"doc":{"_id":"p:106418473048384","_rev":"1-81f88fc9259f481c97f60e8c3a8ab4a3","name":"Seth One","type":"profile","user_id":"106418473048384"}}
],
"total_rows":11,"update_seq":18}%
```

@hideki I emailed you the database.

**Comment by hideki** [comment]
<img width="1326" alt="screen shot 2016-01-29 at 12 01 01 am" src="https://cloud.githubusercontent.com/assets/15889/12670210/70081978-c61b-11e5-8b35-3c8692086ab2.png">

**Comment by sethrosetter** [comment]
@hideki This looks like the a similar issue.

Using ForestDB+Encryption
![device-2016-01-29-002134](https://cloud.githubusercontent.com/assets/1365736/12670574/64bb9470-c61e-11e5-8308-2df0075430b0.png)

Using sqlite
![screenshot 2016-01-29 00 21 47](https://cloud.githubusercontent.com/assets/1365736/12670582/70654244-c61e-11e5-9c73-d219c23fd5f7.png)

**Comment by sethrosetter** [comment]
I think it may be a bug in the app. Although I was not able to get the ForestDB+Encryption database, I deleted the attachment task above and the now the sdfkhasldkjf shows up again, which means it was still in the local data base

After deleting the "photo from dev" task
ForestDB+Encryption
![device-2016-01-29-002607](https://cloud.githubusercontent.com/assets/1365736/12670645/e88e3c3a-c61e-11e5-89d4-29a0634e69fe.png)

sqlite
![screenshot 2016-01-29 00 26 45](https://cloud.githubusercontent.com/assets/1365736/12670655/0439044c-c61f-11e5-8145-00ebaa2dd418.png)

**Comment by pasin** [comment]
@sethrosetter I don't know if it's related to https://github.com/couchbaselabs/ToDoLite-iOS/issues/57.

**Comment by hideki** [comment]
it could be related with https://github.com/couchbase/couchbase-lite-ios/issues/1082

**Comment by hideki** [comment]
@sethrosetter, I need exact steps to reproduce this issue. Please provide the steps?

**Comment by sethrosetter** [comment]
@hideki 

1. Deploy release/1.2.0-forestdb to an API 23 Android Emulator and release/1.2.0-sqlcipher to a device. 
2. Point both to the same sync_gateway
3. Log into the app with different users on the on the emulator and phone
4. Create a list and share it. 
5. Start adding tasks, some with attachments. Verify that they sync
6. Start updating and deleting tasks on the forestdb TodoLite (Long pressing). You may have to repeat this a few times, but you should see the inconsistency eventually.

**Comment by hideki** [comment]
Hi @sethrosetter,

I could reproduce the issue. Thank you for your detailed information.

The task which is next of deleted task is gone on ForestDB. I think this is not replication related. I assume this is same issue of  couchbase/couchbase-lite-ios#1082. As I already cherry-picked the fix into Java. I will update the result.

Left: ForestDB / Right: SQLCipher
Task X was gone when deleted above task.
<img width="1262" alt="screen shot 2016-01-30 at 10 10 45 pm" src="https://cloud.githubusercontent.com/assets/15889/12700589/56a9f3f0-c79e-11e5-8771-40d23744fbae.png">

**Comment by hideki** [comment]
This is duplicated issue with #967 . Once fixed #967, this issue is no longer observable.

Please use 1.2.0-30 for bug fix verification
- http://files.couchbase.com/maven2/com/couchbase/lite/couchbase-lite-android/1.2.0-30/
- http://files.couchbase.com/maven2/com/couchbase/lite/couchbase-lite-android-forestdb/1.2.0-30/
